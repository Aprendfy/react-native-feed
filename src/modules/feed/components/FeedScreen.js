import React, { Component, PropTypes } from 'react';
import { View, Text } from 'react-native';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import ScrollableTabView from 'react-native-scrollable-tab-view';
import FeedList from './FeedList.js';
import CardHeader from '../../../components/feed/CardHeader';
import { colors } from '../../../assets/styles/styles'


import * as actions from './../actions/index';
import * as navActions from '../../navigator/actions/index';

class FeedScreen extends Component {
  constructor(props) {
    super(props);
    this.updateTitle = this.updateTitle.bind(this);
    this.state = {
      title: props.route.params.title,
      tab: props.route.params.tab
    };
  }

  static route = {
    navigationBar: {
      visible: false,
      renderTitle: ({ params }) => {
        const { context = {} } = params;
        const { state = {} } = context;
        return (
          <View
            style={{
              width: '100%',
              height: 48,
              justifyContent: 'center'
            }}
          >
            <Text style={{ fontSize: 18, color: '#FFF' }}>{state.title}</Text>
          </View>
        );
      }
    }
  }

  componentWillMount() {
    const { feedActions, navActions } = this.props;
    feedActions.fetchFeed();
    //navActions.setAlert({ title: 'Do I have a title?', message: 'Oh yes, I do! =)', type: 'info', duration: 6000 });
  }
  componentDidMount() {
    const { navigator } = this.props;
    setTimeout(() => {
      navigator.updateCurrentRouteParams({
        context: this
      });
    }, 600);
  }
  async updateTitle(item) {
    const { tabLabel } = item.ref.props;
    const { navigator } = this.props;
    await this.setState({ title: tabLabel });
    navigator.updateCurrentRouteParams({
      context: this
    });
    // setTimeout(() => {
    //   navigator.updateCurrentRouteParams({
    //     context: this
    //   });
    // }, 600);
  }

  render() {
    return (
      /*<ScrollableTabView
        renderTabBar={() => <View />}
        tabBarBackgroundColor='red'
        tabBarTextStyle={{ fontWeight: '600' }}
        onChangeTab={item => this.updateTitle(item)}
        initialPage={this.state.tab}

      >
        <View style={{ flex: 1 }} tabLabel="Facebook Ads">
          <FeedList index={0} name="Primeira" />
        </View>
        <View style={{ flex: 1 }} tabLabel="Google Ads">
          <FeedList index={1} name="Segunda" />
        </View>
        <View style={{ flex: 1 }} tabLabel="Twitter Ads">
          <FeedList index={2} name="Terceira" />
        </View>
      </ScrollableTabView>
    */
      <View style={{ flex: 1 }}>
        <View style={{ flex: 2 }}>
          <CardHeader
            title="3 Dicas de Publicações no Perfil para Fotógrafos"
            tag="Facebook"
            time="5 minutos"
            level="Iniciante"
            color={colors.categorieFacebook}
            image={{ uri: 'http://i.huffpost.com/gen/3971736/images/o-HAPPY-PEOPLE-facebook.jpg' }}
          />
        </View>
        <View style={{ flex: 3 }}>
          <Text style={{ color: colors.blackPrimary, fontSize: 14, padding: 5 }}>
            Ainda assim, existem dúvidas a respeito de como a complexidade dos estudos efetuados exige a precisão e a definição dos índices pretendidos. Caros amigos, a execução dos pontos do programa apresenta tendências no sentido de aprovar a manutenção de alternativas às soluções ortodoxas. Nunca é demais lembrar o peso e o significado destes problemas, uma vez que a contínua expansão de nossa atividade não pode mais se dissociar do investimento em reciclagem técnica.
            No entanto, não podemos esquecer que a revolução dos costumes auxilia a preparação e a composição das posturas dos órgãos dirigentes com relação às suas atribuições. Do mesmo modo, o entendimento das metas propostas causa impacto indireto na reavaliação das novas proposições. A prática cotidiana prova que a estrutura atual da organização desafia a capacidade de equalização das direções preferenciais no sentido do progresso. Todavia, a necessidade de renovação processual garante a contribuição de um grupo importante na determinação dos métodos utilizados na avaliação de resultados.
          </Text>
        </View>
      </View>
    );
  }
}

FeedScreen.PropTypes = {
  navigator: PropTypes.object
};
FeedScreen.defaultProps = {
  navigator: {}
}

export default connect(
  state => ({
    feedState: state.feed
  }),
  dispatch => ({
    feedActions: bindActionCreators(actions, dispatch),
    navActions: bindActionCreators(navActions, dispatch)
  })
)(FeedScreen);
