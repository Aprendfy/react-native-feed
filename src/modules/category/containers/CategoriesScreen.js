import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { FlatList } from 'react-native';
import { connect } from 'react-redux';
import { ItemList } from '../components/';
import { Router } from '../../../router/routes';
import { fetchCategories } from '../actions';
import { rightSideIcon } from '../../navigator/components/NavigationBarItems';

export class CategoriesScreen extends Component {
  constructor(props) {
    super(props);
    this.onPressCategory = this.onPressCategory.bind(this);
    this.renderItem = this.renderItem.bind(this);
  }

  static route = {
    navigationBar: {
      visible: true,
      title: 'Etiquetas',
      renderRight: rightSideIcon
    }
  }

  componentWillMount() {
    this.props.refreshCategories();
  }

  renderItem({ item, index }) {
    return <ItemList title={item.title} color={item.color} id={index} onPress={this.onPressCategory} />;
  }

  onPressCategory(title, tab, color) {
    this.props.navigator.push(Router.getRoute('feed', { title, tab, color }));
  }

  render() {
    return (
      <FlatList
        renderItem={this.renderItem}
        data={this.props.categories}
        keyExtractor={(item, index) => index}
      />
    );
  }
}

CategoriesScreen.propTypes = {
  categories: PropTypes.array.isRequired,
  refreshCategories: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => {
  const { categories } = state.categories;
  return {
    categories
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    refreshCategories: () => dispatch(fetchCategories())
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(CategoriesScreen);
