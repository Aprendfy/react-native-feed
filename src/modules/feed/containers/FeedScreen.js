import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { NavigationStyles } from '@expo/ex-navigation';

import { FeedList } from '../components/FeedList';
import { colors } from '../../theme/styles';
import { fetchFeed } from '../actions/index';
import { backButtonAndTitle } from '../../navigator/components/NavigationBarItems';

export class FeedScreen extends Component {
  static route = {
    navigationBar: {
      visible: true,
      renderTitle: () => backButtonAndTitle('Facebook')
    },
    styles: {
      ...NavigationStyles.FloatHorizontal
    }
  }

  render() {
    const { getFeeds, feedList } = this.props;
    return (
      <FeedList
        containerStyle={{ flex: 1 }} tabLabel="Facebook"
        index={0} name="Primeira" color={colors.categorieFacebook}
        list={feedList} onEndReached={() => getFeeds('Facebook')}
      />
    );
  }
}

FeedScreen.PropTypes = {
  feedList: PropTypes.array
};

const mapStateToProps = (state) => {
  return {
    feedList: state.feed.feedList
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    getFeeds: bindActionCreators(fetchFeed, dispatch)
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(FeedScreen);
