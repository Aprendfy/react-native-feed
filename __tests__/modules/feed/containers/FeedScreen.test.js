import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import { FeedScreen } from '../../../../src/modules/feed/containers/FeedScreen';

const props = {
  getPosts: jest.fn(),
  route: {
    params: {
      tab: 2,
      title: 'title'
    }
  },
  navigator: {},
  categoryFeeds: []
};

describe('<FeedScreen />', () => {
  it('Should have a snapshot', () => {
    const wrapper = shallow(<FeedScreen {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
