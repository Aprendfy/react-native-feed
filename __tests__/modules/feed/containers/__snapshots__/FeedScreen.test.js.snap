// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeedScreen /> Should have a snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View
    style={
        Object {
            "flex": 1,
          }
    }
>
    <View
        style={
            Object {
                "alignItems": "center",
                "backgroundColor": "#29384D",
                "elevation": 4,
                "flexDirection": "row",
                "height": 64,
                "paddingHorizontal": 0,
                "paddingTop": 20,
              }
        }
    >
        <TouchableOpacity
            activeOpacity={0.2}
            focusedOpacity={0.7}
            onPress={[Function]}
            style={
                Object {
                    "alignItems": "center",
                    "height": 44,
                    "justifyContent": "center",
                    "width": 44,
                  }
            }
        >
            <Icon
                allowFontScaling={false}
                color="white"
                name="md-arrow-back"
                size={24}
            />
        </TouchableOpacity>
        <View
            style={
                Object {
                    "flex": 1,
                  }
            }
        >
            <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                    Object {
                        "color": "#ffffff",
                        "fontSize": 20,
                      }
                }
            >
                title
            </Text>
        </View>
    </View>
    <LoadingSpinner
        spinnerColor="#ffffff"
        spinnerSize="large"
        style={
            Object {
                "backgroundColor": undefined,
              }
        }
    />
</View>,
  "nodes": Array [
    <View
      style={
            Object {
                  "flex": 1,
                }
      }
>
      <View
            style={
                  Object {
                        "alignItems": "center",
                        "backgroundColor": "#29384D",
                        "elevation": 4,
                        "flexDirection": "row",
                        "height": 64,
                        "paddingHorizontal": 0,
                        "paddingTop": 20,
                      }
            }
      >
            <TouchableOpacity
                  activeOpacity={0.2}
                  focusedOpacity={0.7}
                  onPress={[Function]}
                  style={
                        Object {
                              "alignItems": "center",
                              "height": 44,
                              "justifyContent": "center",
                              "width": 44,
                            }
                  }
            >
                  <Icon
                        allowFontScaling={false}
                        color="white"
                        name="md-arrow-back"
                        size={24}
                  />
            </TouchableOpacity>
            <View
                  style={
                        Object {
                              "flex": 1,
                            }
                  }
            >
                  <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        numberOfLines={1}
                        style={
                              Object {
                                    "color": "#ffffff",
                                    "fontSize": 20,
                                  }
                        }
                  >
                        title
                  </Text>
            </View>
      </View>
      <LoadingSpinner
            spinnerColor="#ffffff"
            spinnerSize="large"
            style={
                  Object {
                        "backgroundColor": undefined,
                      }
            }
      />
</View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <FeedScreen
        feedList={Array []}
        getPosts={[Function]}
        navigator={Object {}}
        route={
                Object {
                        "params": Object {
                          "tab": 2,
                          "title": "title",
                        },
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": FeedScreen {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "feedList": Array [],
          "getPosts": [Function],
          "navigator": Object {},
          "route": Object {
            "params": Object {
              "tab": 2,
              "title": "title",
            },
          },
        },
        "refs": Object {},
        "state": Object {
          "backgroundColor": undefined,
          "isLoading": true,
          "tab": 2,
          "title": "title",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <View
          style={
                    Object {
                              "flex": 1,
                            }
          }
>
          <View
                    style={
                              Object {
                                        "alignItems": "center",
                                        "backgroundColor": "#29384D",
                                        "elevation": 4,
                                        "flexDirection": "row",
                                        "height": 64,
                                        "paddingHorizontal": 0,
                                        "paddingTop": 20,
                                      }
                    }
          >
                    <TouchableOpacity
                              activeOpacity={0.2}
                              focusedOpacity={0.7}
                              onPress={[Function]}
                              style={
                                        Object {
                                                  "alignItems": "center",
                                                  "height": 44,
                                                  "justifyContent": "center",
                                                  "width": 44,
                                                }
                              }
                    >
                              <Icon
                                        allowFontScaling={false}
                                        color="white"
                                        name="md-arrow-back"
                                        size={24}
                              />
                    </TouchableOpacity>
                    <View
                              style={
                                        Object {
                                                  "flex": 1,
                                                }
                              }
                    >
                              <Text
                                        accessible={true}
                                        allowFontScaling={true}
                                        ellipsizeMode="tail"
                                        numberOfLines={1}
                                        style={
                                                  Object {
                                                            "color": "#ffffff",
                                                            "fontSize": 20,
                                                          }
                                        }
                              >
                                        title
                              </Text>
                    </View>
          </View>
          <LoadingSpinner
                    spinnerColor="#ffffff"
                    spinnerSize="large"
                    style={
                              Object {
                                        "backgroundColor": undefined,
                                      }
                    }
          />
</View>,
        "_debugID": 2,
        "_renderedOutput": <View
          style={
                    Object {
                              "flex": 1,
                            }
          }
>
          <View
                    style={
                              Object {
                                        "alignItems": "center",
                                        "backgroundColor": "#29384D",
                                        "elevation": 4,
                                        "flexDirection": "row",
                                        "height": 64,
                                        "paddingHorizontal": 0,
                                        "paddingTop": 20,
                                      }
                    }
          >
                    <TouchableOpacity
                              activeOpacity={0.2}
                              focusedOpacity={0.7}
                              onPress={[Function]}
                              style={
                                        Object {
                                                  "alignItems": "center",
                                                  "height": 44,
                                                  "justifyContent": "center",
                                                  "width": 44,
                                                }
                              }
                    >
                              <Icon
                                        allowFontScaling={false}
                                        color="white"
                                        name="md-arrow-back"
                                        size={24}
                              />
                    </TouchableOpacity>
                    <View
                              style={
                                        Object {
                                                  "flex": 1,
                                                }
                              }
                    >
                              <Text
                                        accessible={true}
                                        allowFontScaling={true}
                                        ellipsizeMode="tail"
                                        numberOfLines={1}
                                        style={
                                                  Object {
                                                            "color": "#ffffff",
                                                            "fontSize": 20,
                                                          }
                                        }
                              >
                                        title
                              </Text>
                    </View>
          </View>
          <LoadingSpinner
                    spinnerColor="#ffffff"
                    spinnerSize="large"
                    style={
                              Object {
                                        "backgroundColor": undefined,
                                      }
                    }
          />
</View>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <FeedScreen
    feedList={Array []}
    getPosts={[Function]}
    navigator={Object {}}
    route={
        Object {
            "params": Object {
              "tab": 2,
              "title": "title",
            },
          }
    }
/>,
}
`;
